File: 257.二叉树的所有路径.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> res;
        if(root == NULL)
            return res;
        dfs(root, "", res);
        return res;
    }

    void dfs(TreeNode* root, string s, vector<string>& res){
        s += to_string(root->val);
        if(!root->left && !root->right){
            res.push_back(s);
            return;
        }
        if(root->left)
            dfs(root->left, s + "->", res);
        if(root->right)
            dfs(root->right, s + "->", res);
    }
};

File: 树/257.二叉树的所有路径.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
 var binaryTreePaths = function(root) {
    let res = [];
    let temp = [];

    const dfs = (root) => {
        if (root == null)
            return;
        if (root.left == null && root.right == null)    {
            temp.push(root.val);
            res.push(temp.join(""));
            temp.pop();
            return;
        }
        temp.push(root.val + "->");
        dfs(root.left);
        dfs(root.right);
        temp.pop();
    }

    dfs(root);

    return res;
};